### PR TITLE
feat: Introduce new REPL architecture with persistence and web UI

### PR DESCRIPTION
**Pull request by wesen**
This pull request introduces a comprehensive overhaul of the REPL
functionality, replacing the previous implementations with a new, robust
architecture centered around a persistent REPL service.

The core of this change is a new set of packages (`replapi`,
`replsession`, `repldb`) that provide a stateful, persistent REPL
experience. Sessions are now stored in a local SQLite database, allowing
for inspection, replay, and analysis.

### Key Architectural Changes:

*   **Unified `goja-repl` CLI:** A new command, `goja-repl`, is introduced
    as the single entry point for all REPL interactions. It replaces the
    old `cmd/js-repl` and `cmd/repl` commands, which have been removed.
*   **Persistent Session Service:** The new `replsession` service manages
    the entire lifecycle of a REPL session, including evaluation, state
    management, source code rewriting, and persistence.
*   **SQLite-based Persistence:** All interactions within a REPL session are
    now recorded in a `goja-repl.sqlite` database via the `repldb`
    package. This enables detailed post-session analysis.
*   **Multiple Frontends:** The new service is exposed through several
    interfaces:
    *   An interactive Terminal UI (TUI) via `goja-repl tui`.
    *   An HTTP JSON server via `goja-repl server`.
    *   A new, rich web-based UI for session inspection.

### Web-based Session Inspector ("Essay"):

A significant part of this change is a new web application built with
React and Storybook. This UI, referred to as the "REPL Essay," provides a
detailed, step-by-step visualization of a REPL session. It allows users
to inspect:

*   Code evaluation results.
*   Source code transformations and rewrites.
*   The state of memory bindings.
*   Execution policies and timeouts.
*   The complete persistence history.

### CODE REVIEW

File: pkg/replsession/service.go [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+			MetadataJSON: metadataJSON,
+		}); err != nil {
+			_ = rt.Close(ctx)
+			return nil, errors.Wrap(err, "persist session")
+		}
+	}
+
+	s.mu.Lock()
+	s.sessions[id] = state
+	s.mu.Unlock()
```
[the remark this answers is not in the document]
Comment (wesen) [reply pkg/replsession/service.go:201]: Addressed in 5acdc4e. `CreateSessionWithOptions` now rejects duplicate live explicit session IDs instead of overwriting the existing entry. I added an early explicit-ID check plus a second guarded check before insertion so we also avoid the overwrite race after runtime creation. There is also a regression test (`TestServiceCreateSessionRejectsDuplicateExplicitLiveID`) that verifies the duplicate create fails and the original session remains usable.

File: pkg/replsession/service.go [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+		delete(s.sessions, sessionID)
+	}
+	s.mu.Unlock()
+	if !ok {
+		return ErrSessionNotFound
+	}
+	if state.policy.Persist.Enabled {
+		if err := s.store.DeleteSession(ctx, sessionID, time.Now().UTC()); err != nil {
+			return errors.Wrap(err, "persist session deletion")
+		}
```
[the remark this answers is not in the document]
Comment (wesen) [reply pkg/replsession/service.go:254]: Addressed in 5acdc4e. `DeleteSession` now always closes the runtime after removing the session from the live map, even if durable deletion fails. If both the persistence delete and runtime close fail, the returned error now includes both failure paths. I also added a regression test (`TestServiceDeleteSessionClosesRuntimeWhenPersistenceDeleteFails`) to verify the runtime close hook still runs when the store delete returns an error.

**Comment by wesen** [comment]
@codex review
